llvm/lib/Target/AArch64/AArch64MachineFunctionInfo.cpp

```diff
@@ -82,6 +83,25 @@ static bool ShouldSignWithBKey(const Function &F, const AArch64Subtarget &STI) {
   return Key == "b_key";
 }
 
+static bool hasELFSignedGOTHelper(const Function &F,
+                                  const AArch64Subtarget *STI) {
+  if (!Triple(STI->getTargetTriple()).isOSBinFormatELF())
+    return false;
+  const Module *M = F.getParent();
+  uint64_t PAuthABIPlatform = -1;
+  if (const auto *PAP = mdconst::extract_or_null<ConstantInt>(
+          M->getModuleFlag("aarch64-elf-pauthabi-platform")))
+    PAuthABIPlatform = PAP->getZExtValue();
+  if (PAuthABIPlatform != ELF::AARCH64_PAUTH_PLATFORM_LLVM_LINUX)
+    return false;
```

Comment:
```suggestion
  const auto *PAP = mdconst::extract_or_null<ConstantInt>(
          M->getModuleFlag("aarch64-elf-pauthabi-platform"));
  if (!PAP || PAP->getZExtValue() != ELF::AARCH64_PAUTH_PLATFORM_LLVM_LINUX)
      return false;
```
This avoids the magic -1 and makes the logic easier to follow, imho.

Reply:
I agree, it's much better, thanks! Applied your suggestion in 4c85d6c31cfc89e8543d4f8d78c9cd82dd33ca37